src/resolver.ts

```diff
@@ -23,3 +38,53 @@ export const resolvers = {
     },
   },
 };
+
+function thereIsNumber(str: string) {
+  return isNaN(parseFloat(str));
+}
+
+function thereIsLetter(str: string) {
+  const alf = [
+    'a',
+    'b',
+    'c',
+    'd',
+    'e',
+    'f',
+    'g',
+    'h',
+    'i',
+    'j',
+    'k',
+    'l',
+    'm',
+    'n',
+    'o',
+    'p',
+    'q',
+    'r',
+    's',
+    't',
+    'u',
+    'v',
+    'w',
+    'x',
+    'y',
+    'z',
+  ];
+  let def = false;
+  alf.forEach(function (letter) {
+    if (str.includes(letter)) {
+      def = true;
+    }
+  });
+  return def;
+}
+
+async function emailAlreadyUsed(str: string) {
+  const userRepository = AppDataSource.getRepository(User);
+  const user = await userRepository.findOneBy({
+    email: str,
+  });
+  return user != null;
```

Comment:
```suggestion
  return !user;
```

No geral é legal evitar comparações usando `==` ou `!=`, pq no Javascript essa comparação leva em conta apenas a falsidade ou não dos valores, não leva em conta do tipo. O ideal é sempre tentar usar `!==` e `===`. Mais detalhes [aqui](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness).